### DISCUSSION TITLE
Default write behaviour

### DISCUSSION THREAD

**Discussion by sam-carter-mcgrand** [scalameta/nvim-metals#681] [Q&A] [answered]
Forgive me if I'm missing something obvious, but is there a way to change the [default save](https://github.com/scalameta/nvim-metals/discussions/279#discussioncomment-1752011) behaviour?
I don't always want to compile immediately after `:w`, especially in larger projects

**Comment by ckipp01** [comment, marked as answer]
Hey @sam-carter-mcgrand, not really without really hacking something. In LSP the didSave event triggers a save in Metals, so this is the expected behavior.

**Reply by sam-carter-mcgrand** [reply]
Ah thanks @ckipp01, I'll dig around and see if I can hack something that works for me
